core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -217,7 +240,7 @@ Kokkos::HIP::size_type *HIPInternal::scratch_space(const std::size_t size) {
 Kokkos::HIP::size_type *HIPInternal::scratch_flags(const std::size_t size) {
   if (verify_is_initialized("scratch_flags") &&
       m_scratchFlagsCount < scratch_count(size)) {
-    Kokkos::HIPSpace mem_space;
+    auto mem_space = Kokkos::HIPSpace::impl_create(m_hipDev);
```

Comment:
No stream? You will need that for `hipMallocAsync`, right?

Reply:
Right now `hipMallocAsync` isn't used and `HIPSpace` doesn't have a `hipStream m_stream` member variable.

Reply:
I'm fine with either adding it here or the pull request adding `hipMallocAsync` support.

Reply:
Looks like this is already in the works https://github.com/kokkos/kokkos/pull/7324. There has been action on that PR, so maybe we wait on it to merge and then I rebase and make my necessary changes.